### ISSUE TITLE
project configuration is set disable build, however dotnet run still build

### ISSUE THREAD

**Issue by hochun836** [dotnet/sdk#31692]
### Describe the bug
The solution structure is the following:
`DemoNetBuild.sln`
- `DemoConsole.csproj`
- `DemoUtil.csproj`
- `DemoData.csproj`
- `DemoCore.csproj`

The `DemoConsole.csproj` depends on
- `DemoUtil.csproj`
- `DemoData.csproj`
- `DemoCore.csproj`

In visual stuido, I have set `Debug` configuration
![image](https://user-images.githubusercontent.com/13325100/231049051-4d0b5417-b88e-4fa2-b5f7-88abe9f78750.png)

When run `DemoConsole` in visual stuido, I see 3 skipped
![image](https://user-images.githubusercontent.com/13325100/231049212-cb2a4bc7-bb66-4374-a40c-56d64de1dc0f.png)

But when I run by `dotnet run DemoConsole -c Debug`, all projects still be built.

I expect dotnet cli like visual studio to skip some projects.

### To Reproduce
```
mkdir DemoNetBuild && cd DemoNetBuild
dotnet new console -o DemoConsole
dotnet new classlib -o DemoUtil
dotnet new classlib -o DemoData
dotnet new classlib -o DemoCore
dotnet new sln
dotnet sln add DemoConsole DemoUtil DemoData DemoCore
dotnet add DemoConsole reference DemoUtil DemoData DemoCore
dotnet add DemoUtil reference DemoData DemoCore
dotnet add DemoData reference DemoCore
```

### Further technical details
- dotnet 6
- vs 2022

**Comment by nagilson** [comment]
I dont believe there are plans to support a feature like this as you can just run build and specify the console project only. But we can discuss this with our PM.
